src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,229 @@
+        boolean isDone = false;
+        Scanner in = new Scanner(System.in);
+
+        printWelcomeMessage();
+
+        do {
+            // Read inputs and split them into substrings
+            String[] userInputs = readUserInput(in);
+            String command = userInputs[0];
+
+            switch (command) {
+            case "bye":
+                isDone = true;
+                printExitMessage();
+                break;
+            case "list":
+                printTaskList(taskList);
+                break;
+            case "done":
+                try {
+                    executeDone(userInputs);
+                } catch (Exception exception) {
+                    printErrorMessage();
+                }
+                break;
+            case "todo":
+                try {
+                    executeTodo(userInputs);
+                } catch (EmptyDescriptionException exception) {
+                    printEmptyDescriptionErrorMessage();
+                } catch (Exception exception) {
+                    printErrorMessage();
+                }
+                break;
+            case "deadline":
+                try {
+                    executeDeadline(userInputs);
+                } catch (EmptyTimeFieldException exception) {
+                    printEmptyTimeFieldErrorMessage();
+                } catch (EmptyDescriptionException exception) {
+                    printEmptyDescriptionErrorMessage();
+                } catch (Exception exception) {
+                    printErrorMessage();
+                }
+                break;
+            case "event":
+                try {
+                    executeEvent(userInputs);
+                } catch (EmptyTimeFieldException exception) {
+                    printEmptyTimeFieldErrorMessage();
+                } catch (EmptyDescriptionException exception) {
+                    printEmptyDescriptionErrorMessage();
+                } catch (Exception exception) {
+                    printErrorMessage();
+                }
+                break;
+            default:
+                printErrorMessage();
+                break;
+            }
+        } while (!isDone);
+    }
+
+
+
+    private static void executeEvent(String[] userInputs) throws EmptyDescriptionException, EmptyTimeFieldException {
+        if (userInputs.length < 2) {
+                throw new EmptyDescriptionException();
+        }
+        if (!userInputs[1].contains("/at")) {
+            throw new EmptyTimeFieldException();
+        }
+        String eventDescription = userInputs[1]
+                .substring(0, userInputs[1].indexOf("/"))
+                .strip();
+        if (eventDescription.isBlank()) {
+            throw new EmptyDescriptionException();
+        }
+        String eventAt = userInputs[1].substring((userInputs[1]
+                        .indexOf("/") + 3))
+                .strip();
+        taskList[Task.getNumOfTasks()] = new Event(eventDescription, eventAt);
```

Comment:
As the taskList is an array with a fixed length, it would be better if a solution could be added to prevent index out of bound error for the array here.